### PR TITLE
support distinguish configuration of different environments by environment variables

### LINKED ISSUE #1123: feature: Does apisix support distinguish configuration of different environments by environment variables ?

### PR DESCRIPTION
**Pull request by qiujiayu**
Fix #1123

### CODE REVIEW

**Comment by membphis** [comment]
@qiujiayu you should use `git rebase` to update your working branch.

**Comment by membphis** [comment]
please pay attention at the  Travis outputs: https://travis-ci.org/apache/incubator-apisix/jobs/651568717#L682

we need to add a test case for this feature, you can take a look at another PR: https://github.com/apache/incubator-apisix/pull/1133

**Comment by membphis** [comment]
@qiujiayu https://github.com/apache/incubator-apisix/pull/1133 had been merged, you can add new test case after you rebase your branch.

File: lua/apisix/core/config_local.lua [hunk 1]
```diff
@@ -28,6 +30,7 @@ local _M = {
 }
 
 
+
```
Comment (membphis) [inline lua/apisix/core/config_local.lua]: useless blank line, please remove it

File: lua/apisix/core/profile.lua [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local _M = {
+    version = 0.2,
```
Comment (membphis) [inline lua/apisix/core/profile.lua]: first version, it should be `0.1`

File: lua/apisix/core/profile.lua [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    profile = os.getenv("APISIX_PROFILE")
+}
+
+
+function _M.build_yaml_config_file(self, file_path)
+    if self.profile then
```
Comment (membphis) [inline lua/apisix/core/profile.lua]: here is a better style:

```lua
if self.profile then
    return ... ...
end

return ... ...
```

**Comment by membphis** [comment]
@qiujiayu please rebase your branch, I updated the code style check tools.

**Comment by membphis** [comment]
@qiujiayu https://apisix.dev/d/1-apache-apisix/10

I think you can join this event.

**Comment by qiujiayu** [comment]
> @qiujiayu https://apisix.dev/d/1-apache-apisix/10
> 
> I think you can join this event.

joined

**Comment by membphis** [comment]
@qiujiayu I think you can create a new PR with your commits if you do not know how to fix the wrong commit log.

**Pull request closed without merging**